hotham/src/resources/xr_context.rs

```diff
@@ -354,45 +396,24 @@ pub(crate) fn create_xr_session(
     .unwrap())
 }
 
-pub(crate) fn create_xr_instance() -> anyhow::Result<(xr::Instance, xr::SystemId)> {
-    let xr_entry = xr::Entry::load()?;
-    let xr_app_info = openxr::ApplicationInfo {
-        application_name: "Hotham Asteroid",
-        application_version: 1,
-        engine_name: "Hotham",
-        engine_version: 1,
-    };
-    let mut required_extensions = xr::ExtensionSet::default();
-    // required_extensions.khr_vulkan_enable2 = true; // TODO: Should we use enable 2 for the simulator..?
-    required_extensions.khr_vulkan_enable = true; // TODO: Should we use enable 2 for the simulator..?
-
-    #[cfg(target_os = "android")]
-    {
-        required_extensions.khr_android_create_instance = true;
-        xr_entry.initialize_android_loader()?;
-    }
-
-    println!(
-        "Available extensions: {:?}",
-        xr_entry.enumerate_extensions()?
-    );
-
-    let instance = xr_entry.create_instance(&xr_app_info, &required_extensions, &[])?;
-    let system = instance.system(xr::FormFactor::HEAD_MOUNTED_DISPLAY)?;
-    Ok((instance, system))
-}
-
-pub(crate) fn create_xr_instance_from_path(
-    path: &std::path::Path,
+pub(crate) fn create_xr_instance(
+    path: Option<&std::path::Path>,
+    application_name: Option<&str>,
+    application_version: Option<u32>,
+    required_extensions: Option<&xr::ExtensionSet>,
 ) -> anyhow::Result<(xr::Instance, xr::SystemId)> {
-    let xr_entry = xr::Entry::load_from(path)?;
+    let xr_entry = if let Some(path) = path {
+        xr::Entry::load_from(path)?
+    } else {
+        xr::Entry::load()?
+    };
     let xr_app_info = openxr::ApplicationInfo {
-        application_name: "Hotham Asteroid",
-        application_version: 1,
+        application_name: application_name.unwrap_or("Hotham Asteroid"),
```

Comment:
nit: Hotham Asteroid is a very old reference to the original "Asteroid" demo I created. "Hotham Application" is a better substitute.